### DISCUSSION TITLE
Handling disabled GPS on Android

### DISCUSSION THREAD

**Discussion by CptFabulouso** [react-native-maps/react-native-maps#4516] [Feature requests]
### Summary

There is no message or anything if user wants to get their location, but they've turned the gps module off (not talking about permission). I'd expect they are displayed the dialog to turn the gps module on (like in google maps app) (like the lib [react-native-android-location-enabler](https://github.com/Richou/react-native-android-location-enabler) allows to). If you don't want to handle this, can we have some callback on the "user location" icon press, where we can handle it? 

### Reproducible sample code

```TSX
More a feature request than a bug, there's nothing extra to share, just basic configuration.
```

### Steps to reproduce

- Turn gps module off
- press "user location" icon

### Expected result

Display prompt to enable GPS module

### Actual result

Nothing happens

### React Native Maps Version

1.3.2

### What platforms are you seeing the problem on?

Android

### React Native Version

0.69.1

### What version of Expo are you using?

Not using Expo

### Device(s)

Samsung S22 (Android 12)

**Comment by Bartosz-L** [comment]
Write your own location button and check if user gave permission. If not, do something like alert. In my case I'm invoking askForLocation from the button on map.

```Typescript
import Geolocation from '@react-native-community/geolocation';
import { t } from 'i18next';
import { Alert } from 'react-native';
import MapView, { Region } from 'react-native-maps';

interface AskForLocationProps {
  region: Region;
  mapRef: React.RefObject<MapView>;
}

export const askForLocation = ({ region, mapRef }: AskForLocationProps) => {
  Geolocation.getCurrentPosition(
    (success) => {
      mapRef.current?.animateToRegion({
        ...region,
        latitude: success.coords.latitude,
        longitude: success.coords.longitude,
      });
    },
    (error) => {
      if (error.PERMISSION_DENIED === 1) {
        Alert.alert(t('st-cm-no-access-to-user-location'), t('st-cm-no-access-to-user-location-description'));
      }
    },
  );
};
```

**Comment by monholm** [comment]
Hi @CptFabulouso 👋

I'm against having `react-native-maps` handle the fact that it's turned off (e.g. showing a dialog), but having a callback function for the `onPress` seems valuable. If you are able to, feel free to provide a PR with the functionality, and I'll be happy to help getting it merged. 👍
